packages/freeswitch/src/eventParser/parseEventData.ts

```diff
@@ -0,0 +1,168 @@
+		'Channel-Read-Codec-Rate': codecReadRate,
+		'Channel-Write-Codec-Name': codecWriteName,
+		'Channel-Write-Codec-Rate': codecWriteRate,
+
+		...rawEventData
+	} = eventData;
+
+	if (!channelUniqueId || !sequenceStr) {
+		logger.error({ msg: 'Channel Event is missing either the Unique-ID or Event-Sequence', eventData });
+		return;
+	}
+
+	const sequence = parseInt(sequenceStr);
+	if (!sequence || typeof sequence !== 'number' || !Number.isInteger(sequence)) {
+		logger.error({ msg: 'Failed to parse Event-Sequence', eventData });
+		return;
+	}
+
+	const callUniqueId = parseEventCallId(eventData) || channelUniqueId;
+	const channelUsername = parseChannelUsername(channelName);
+	const firedAt = parseTimestamp(timestamp) || new Date();
+
+	const callerLeg = parseEventLeg('Caller', eventData);
+	const otherLeg = parseEventLeg('Other-Leg', eventData);
+	const bridgeUniqueIds = [bridgeA, bridgeB].filter((bridgeId) => bridgeId) as string[];
+
+	const legs: IFreeSwitchChannelEvent['legs'] = {
+		...(callerLeg?.uniqueId && { [callerLeg.uniqueId]: callerLeg }),
+		...(otherLeg?.uniqueId && { [otherLeg.uniqueId]: otherLeg }),
+	};
+
+	const variables = filterStringList(
+		eventData,
+		(key) => key.startsWith('variable_'),
+		([key, value]) => {
+			return [key.replace('variable_', ''), value || ''];
+		},
+	);
+	const metadata = filterStringList(eventData, (key) => isMetadata(key));
+	const unusedRawData = filterStringList(rawEventData, (key) => {
+		if (isMetadata(key)) {
+			return false;
+		}
+		if (key.startsWith('variable_')) {
+			return false;
+		}
+
+		for (const { legName } of Object.values(legs)) {
+			if (key.startsWith(`${legName}-`)) {
+				return false;
+			}
+		}
+
+		if (otherLeg && key === 'Other-Type') {
+			return false;
+		}
+
+		if (key === 'Channel-Call-UUID') {
+			return rawEventData['Channel-Call-UUID'] !== callUniqueId;
+		}
+
+		return true;
+	});
+
+	const event: Omit<IFreeSwitchChannelEvent, '_id' | '_updatedAt'> = {
+		channelUniqueId,
+		eventName,
+		sequence,
+		firedAt,
+		receivedAt: new Date(),
+		callUniqueId,
+		channelName,
+		channelState,
+		channelStateNumber,
+		channelCallStateNumber,
+		channelCallState,
+
+		originalChannelCallState,
+		channelUsername,
+		answerState,
+		callDirection,
+		channelHitDialplan,
+		hangupCause,
+
+		...(bridgeUniqueIds.length && { bridgeUniqueIds }),
+		bridgedTo,
+		legs,
+		metadata: filterOutEmptyValues(metadata),
+		...((Object.keys(variables).length && { variables }) as { variables: IFreeSwitchChannelEvent['variables'] } | undefined),
+		raw: filterOutEmptyValues(unusedRawData),
+
+		codecs: {
+			...{
+				read: {
+					...{
+						name: codecReadName,
+						rate: codecReadRate,
+					},
+				},
+				write: {
+					...{
+						nme: codecWriteName,
+						rate: codecWriteRate,
+					},
+				},
+			},
+		},
+
+		presenceCallDirection,
+		channelPresenceId,
+	};
+
+	const filteredEvent = {
+		...(filterOutMissingData(event) as typeof event),
```

Comment:
filterOutMissingData should just give the same type back instead of requiring this cast.